core/sdk/src/api/storage_property.rs

```diff
@@ -547,6 +548,11 @@ impl StorageProperty {
             StorageProperty::AudioDescriptionEnabled => PROPERTY_AUDIO_DESCRIPTION_ENABLED,
             StorageProperty::PreferredAudioLanguages => PROPERTY_PREFERRED_AUDIO_LANGUAGES,
             StorageProperty::CCPreferredLanguages => PROPERTY_CC_PREFERRED_LANGUAGES,
+            StorageProperty::CustomProperty(namespace, key) => PropertyData {
+                namespace: Box::leak(namespace.clone()),
```

Comment:
Where exactly does this memory get deallocated ?

Reply:
@VinodsathyaseelanI used Box::leak to extend the lifetime of a String to 'static because the other storage properties are all static strings. This ensures consistency in how the properties are managed. However, I understand this approach leads to intentional memory leaks, as the memory will not be deallocated. If you have suggestions for a better approach to achieve this while avoiding memory leaks, I'd be happy to consider them.

Reply:
You may need to reclaim the leaked entries from `PropertyData` back to Box right after completing the usage. 

Refer : Box::from_raw()